### PR TITLE
docs(upgrading): add "Migrating from Vite" guide

### PR DESCRIPTION
**Pull request by IGassmann**
### What?

This adds to the docs a guide for migrating a Vite application to Next.js. It also contains a section that lists the reasons why one would want to switch from Vite to Next.js.

### Why?

This guide is helpful for developers that want to switch to Next.js but don't know how difficult that would be and how to do it.   The "Why?" list is also helpful to developers who haven't yet been convinced to switch or need help formulating the reasons to switch to decision-makers.

This guide was [first published on the Inngest blog](https://www.inngest.com/blog/migrating-from-vite-to-nextjs). To our own surprise, we were able to [migrate from Vite to Next.js](https://github.com/inngest/inngest/pull/479) our [Dev Server app](https://github.com/inngest/inngest/tree/main#the-inngest-dev-server) in less than a day, which motivated us to share our learnings with the community.

### How?

The guide aims to have the minimum number of steps to get a working Next.js application so that
the reader can then adopt Next.js features incrementally. This means the migration of the router isn't part of this guide. However, this would be something valuable to document at a later time so that the reader can also learn how to migrate to the App Router and get all the Next.js benefits.

### CODE REVIEW

**Comment by leerob** [comment]
Made some suggestions here: https://github.com/vercel/next.js/commit/ec31bf0c5dbdf2254e9a2ebf51640f04ec7d937f

**Comment by IGassmann** [comment]
Looks great! Just applied the edits in [`16d1c52` (#53214)](https://github.com/vercel/next.js/pull/53214/commits/16d1c52802a8f28d3e69c66d4d2b2c6a8d3c2284)

**Review by leerob — APPROVED** [review]

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 16d1c52802a8f28d3e69c66d4d2b2c6a8d3c2284

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 16d1c52802a8f28d3e69c66d4d2b2c6a8d3c2284

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: docs/02-app/01-building-your-application/08-upgrading/03-from-vite.mdx [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+do next:
+
+- Migrate from React Router to the [Next.js App Router](/docs/app/building-your-application/routing) to get:
+  - Automatic code splitting
+  - [Streaming Server-Rendering](/docs/app/building-your-application/routing/loading-ui-and-streaming)
+  - [React Server Components](/docs/getting-started/react-essentials#server-components)
+- [Optimize images with the `<Image>` component](/docs/app/building-your-application/optimizing/images)
+- [Optimize fonts with `next/font`](/docs/app/building-your-application/optimizing/fonts)
+- [Optimize third-party scripts with the `<Script>` component](/docs/app/building-your-application/optimizing/scripts)
+- [Update your ESLint configuration to support Next.js rules](/docs/app/building-your-application/configuring/eslint)
```
Comment (leerob) [inline docs/02-app/01-building-your-application/08-upgrading/03-from-vite.mdx:561]:
```suggestion
- [Update your ESLint configuration to support Next.js rules](/docs/app/building-your-application/configuring/eslint)

```

**Review by leerob — DISMISSED** [review]

**Review by leerob — APPROVED** [review]

**Pull request merged**